### ISSUE TITLE
[BUG] Working with focus is not possible

### ISSUE THREAD

**Issue by gerken-tss** [the-airbender/quill_html_editor#51]
**Describe the bug**
I have a `QuillHtmlEditor` widget and want the send button to change its color when the focus of the editor changes.

I have tried it in two ways:
* Using the `onFocusChanged` argument of the widget and thus reacting on it to change
* Using the `hasFocus` method which is defined on the controller

Both of the variants don't seem to work.

* `onFocusChanged` is triggered when the editor is created and instantly gives `true` although the editor is not focused.
* `hasFocus` returns `0` initially and then always `null`

In the second variant I used it like this:

```
FutureBuilder<int>(
  future: _textEditingController.hasFocus(),
  builder: (BuildContext context, AsyncSnapshot<int> snapshot) {
    return Icon(
      Icons.send,
      color: snapshot.data == null ? Colors.black45: Theme.of(context).primaryColor,
    );
  },
);
```

I was able to reproduce this on a physical iPhone X and an Android emulator. Haven't tried with a physical Android device.

It's also possible that I misinterpret the intended way to work with focus in this package.

Maybe it should be possible to work with a `FocusNode`, Flutter's built-in mechanism for these kinds of issues.

**Flutter Doctor**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.7.3, on macOS 13.1 22C65 darwin-arm64, locale
    en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.2.3)
[✓] Connected device (5 available)
[✓] HTTP Host Availability
```

**Comment by the-airbender** [comment]
Hi @gerken-tss 

Thank you for reporting this bug.

I have tested it on an iOS devices and able to reproduce it.  The focus is not getting triggered sometimes, will fix it and move it in next release.

Thanks :)

**Comment by the-airbender** [comment]
Hi @gerken-tss ,

I have fixed this issue and released in the ver 2.1.9. Please check it out and share the feedback.

Thanks :)

**Comment by antonydieu** [comment]
quill_html_editor 2.2.6 
flutter: has focus false
flutter: has focus true
Not correctly
